mesonbuild/minstall.py

```diff
@@ -246,14 +248,36 @@ def restore_selinux_contexts() -> None:
               'Standard output:', out,
               'Standard error:', err, sep='\n')
 
-def apply_ldconfig() -> None:
+def apply_ldconfig(dm: DirMaker) -> None:
     '''
     Apply ldconfig to update the ld.so.cache.
     '''
     if not shutil.which('ldconfig'):
         # If we don't have ldconfig, failure is ignored quietly.
         return
 
+    # Get the list of directories where ldconfig looks for its cache
+    proc, out, err = Popen_safe(['ldconfig', '-v', '-N', '-X'])
```

Comment:
This is broken on non-Linux at least because -N and -X aren't valid options on e.g. openbsd, freebsd, netbsd. They all do have `-v`, but I haven't checked their output and I'm nervous that this human-readable output is going to have significant formatting differences...

Reply:
Pfff, sad :(

What I liked with this PR is it also prevent wasting time when the project does not install shared libraries.

I guess for now we could just check `os.geteuid() == 0`, or do you have a smarter idea?

Reply:
I'm pretty sure at the time of the original ldconfig implementation I tried and failed to think of anything.

Checking for root is also "maybe technically wrong -- what if the user has ldconfig set up to index non-root directories" but that is getting pretty niche so idk.

If you want to avoid wasting time when the project does not install shared libraries, you could make ldconfig be an attribute set by `library(..., install: true)`?